src/controller/SetUpCodePairer.cpp

```diff
@@ -405,9 +406,17 @@ void SetUpCodePairer::NotifyCommissionableDeviceDiscovered(const Dnssd::CommonRe
     ConnectToDiscoveredDevice();
 }
 
-void SetUpCodePairer::CommissionerShuttingDown()
+bool SetUpCodePairer::StopPairing(NodeId remoteId)
 {
+    if ((remoteId != kUndefinedNodeId && remoteId != mRemoteId) || mRemoteId == kUndefinedNodeId)
+    {
+        return false; // Not discovering the specified node (or any node)
+    }
+
+    PASEEstablishmentComplete();
     ResetDiscoveryState();
+    mRemoteId = kUndefinedNodeId;
```

Comment:
Can we move `mRemoteId = kUndefinedNodeId` into `ResetDiscoveryState` ?
It seems that the only reason why it can't is because it is used in `PairDevice` after having set `mRemoteId` to something but I assume that we can shuffle the code a little bit ?

Reply:
We could, but it seemed to fit more with `mConnectionType`, `mDiscoveryType` and `mSetUpPINCode` which also come from `Pair...` arguments and we don't clear those either.